node/modules/storageminer.go

```diff
@@ -357,76 +356,187 @@ func SectorStorage(mctx helpers.MetricsCtx, lc fx.Lifecycle, lstor *paths.Local,
+		case errors.Is(err, api.ErrF3Disabled):
+			log.Errorw("Cannot await F3 participation lease expiry as F3 is disabled.", "err", err)
+			return xerrors.Errorf("awaiting F3 participation lease expiry: %w", err)
+		case err != nil:
+			if p.backoff.Attempt() > float64(p.maxCheckProgressAttempts) {
+				log.Errorw("Too many failures while attempting to check F3  progress. Restarting participation.", "attempts", p.backoff.Attempt(), "err", err)
+				return nil
 			}
+			log.Errorw("Failed to check F3 progress while awaiting lease expiry. Retrying after backoff.", "attempts", p.backoff.Attempt(), "backoff", p.backoff.Duration(), "err", err)
+			p.backOff(ctx)
+		case progress.Instance+2 >= lease.ValidityTerm:
+			log.Infof("F3 progressed (%d) to within two instances of lease expiry (%d). Restarting participation.", progress.Instance, lease.ValidityTerm)
+			return nil
+		default:
+			remainingInstanceLease := lease.ValidityTerm - progress.Instance
+			log.Debugf("F3 participation lease is valid for further %d instances. Recking after %s.", remainingInstanceLease, p.checkProgressInterval)
+			p.backOffFor(ctx, p.checkProgressInterval)
+		}
+	}
+	return ctx.Err()
+}
 
-			// we have succeeded in giving a lease, reset the backoff
-			b.Reset()
+func (p *f3Participator) backOff(ctx context.Context) {
+	p.backOffFor(ctx, p.backoff.Duration())
+}
 
-			oldLease = newLease
-			// wait for the half of the lease time and then refresh
-			timer.Reset(leaseTime / 2)
-			select {
-			case <-ctx.Done():
-				return
-			case <-timer.C:
-			}
+func (p *f3Participator) backOffFor(ctx context.Context, d time.Duration) {
+	if !p.timer.Stop() {
+		<-p.timer.C
+	}
+	p.timer.Reset(d)
+	select {
+	case <-ctx.Done():
+		return
+	case <-p.timer.C:
+	}
+}
+
+func (p *f3Participator) stop() {
+	p.timer.Stop()
+}
+
+func F3Participation(mctx helpers.MetricsCtx, lc fx.Lifecycle, node v1api.FullNode, participant dtypes.MinerAddress) error {
+	ctx := helpers.LifecycleCtx(mctx, lc)
+	const (
+		// maxCheckProgressAttempts defines the maximum number of failed attempts
+		// before we abandon the current lease and restart the participation process.
+		//
+		// The default backoff takes 12 attempts to reach a maximum delay of 1 minute.
+		// Allowing for 13 failures results in approximately 2 minutes of backoff since
+		// the lease was granted. Given a lease validity of up to 5 instances, this means
+		// we would give up on checking the lease during its mid-validity period;
+		// typically when we would try to renew the participation ticket. Hence, the value
+		// to 13.
+		checkProgressMaxAttempts = 13
+		// checkProgressInterval defines the duration between progress checks in normal operation mode.
+		// This interval is used when there are no errors in retrieving the current progress.
+		checkProgressInterval = 10 * time.Second
+		// leaseTerm The number of instances the miner will attempt to lease from nodes.
+		leaseTerm = 5
+	)
+
+	go func() {
```

Comment:
It would be really nice (especially for testing) if we could wait for this to stop on exit (e.g., with a waitgroup plus an exit hook).